bundles/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComMessageFactory.java

```diff
@@ -93,36 +38,12 @@
 
 	
 	public static RFXComMessageInterface getMessageInterface(PacketType packetType) throws RFXComException {
-		
-		try {
-			String className = messageClasses.get(packetType);
-			Class<?> cl = Class.forName(classUrl + className);
-			return (RFXComMessageInterface) cl.newInstance();
-			
-		} catch (ClassNotFoundException e) {
-			throw new RFXComException("Message " + packetType + " not implemented", e);
-			
-		} catch (Exception e) {
-			throw new RFXComException(e);
-		} 
+		return packetType.createMessage();
 	}
 	
 	public static RFXComMessageInterface getMessageInterface(byte[] packet) throws RFXComException {
-		
-		PacketType packetType = getPacketType(packet[1]);
-		
-		try {
-			String className = messageClasses.get(packetType);
-			Class<?> cl = Class.forName(classUrl + className);
-			Constructor<?> c = cl.getConstructor(byte[].class);
-			return (RFXComMessageInterface) c.newInstance(packet);
-		
-		} catch (ClassNotFoundException e) {
-			throw new RFXComException("Message " + packetType + " not implemented", e);
-			
-		} catch (Exception e) {
-			throw new RFXComException(e);
-		} 
+		final PacketType packetType = getPacketType(packet[1]);
+		return getMessageInterface(packetType);
```

Comment:
Hi @juri8 

This doesn't work similarly than previous code. Now it return object with default values not object based on data (byte[] packet).